### DISCUSSION TITLE
Starlette backend does not appear to be working

### DISCUSSION THREAD

**Discussion by limx0** [perspective-dev/perspective#1957] [Q&A] [answered]
## Bug Report

I'm trying to migrate some custom FastAPI code over to the new perspective Starlette backend, but am not seeing any data updates or any evidence of the tables in a simple html frontend.

### Steps to Reproduce:
1. Run the example [starlette backend](https://github.com/finos/perspective/blob/master/examples/python-starlette/server.py)
2. Create a simple perspective JS frontend:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
        <script src="https://cdn.jsdelivr.net/npm/@finos/perspective"></script>
        <script src="https://cdn.jsdelivr.net/npm/@finos/perspective-viewer"></script>
        <script src="https://cdn.jsdelivr.net/npm/@finos/perspective-viewer-datagrid"></script>
        <script src="https://cdn.jsdelivr.net/npm/@finos/perspective-viewer-d3fc"></script>
</head>
<body>
    <script>
      window.addEventListener("DOMContentLoaded", async function () {
        const websocket = perspective.websocket("ws://localhost:8080/websocket");
        const table = await websocket.open_table("data_source_one");
        const viewer = document.getElementById("viewer");
        await viewer.load(table);
      });
    </script>
    <perspective-viewer id="viewer"></perspective-viewer>
</body>
</html>
```

### Expected Result:
Can see datagrid with data in browser

### Actual Result:
Empty datagrid without any schema, no exceptions. 

### Environment:
Backend
- `perspective-python==1.6.3`
- `fastapi==0.79.1`
- `starlette==0.19.1`

Frontend:
- CDN

### Additional Context:
I can see the websocket connection being established, and data in the backend. The frontend js code connects to the socket and seemingly loads the table, but the viewer does not update.

**Comment by timkpaine** [comment, marked as answer]
I don't see any issues on latest release other than that the `perspective-viewer` element is 0 sized. modifying your `index.html` to match the [examples in the examples folder](https://github.com/finos/perspective/blob/master/examples/python-tornado/index.html#L25):
```
<style>
    perspective-viewer {
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
    }
</style>
```

and I see everything working as normal. do you see the same?

![tmp](https://user-images.githubusercontent.com/3105306/187041633-2ab7f565-e903-4c37-b3fe-e7690a1af75f.gif)

**Reply by limx0** [reply]
Yep confirming I see the same after adding the style, thanks @timkpaine!
